packages/smooth_app/lib/pages/onboarding/next_button.dart

```diff
@@ -0,0 +1,59 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:smooth_app/data_models/user_preferences.dart';
+import 'package:smooth_app/pages/onboarding/onboarding_flow_navigator.dart';
+import 'package:smooth_app/themes/smooth_theme.dart';
+import 'package:smooth_ui_library/util/ui_helpers.dart';
+
+/// Next button showed at the bottom of the onboarding flow.
+class NextButton extends StatelessWidget {
+  const NextButton(this.currentPage);
+
+  final OnboardingPage currentPage;
+
+  // Side padding is 8% of total width.
+  double sidePadding(double screenWidth) => screenWidth * 0.08;
+
+  @override
+  Widget build(BuildContext context) {
+    final Size screenSize = MediaQuery.of(context).size;
+    final AppLocalizations appLocalizations = AppLocalizations.of(context)!;
+    return Container(
+      color: Theme.of(context).appBarTheme.backgroundColor,
+      padding: EdgeInsets.symmetric(
+        vertical: VERY_LARGE_SPACE,
+        horizontal: sidePadding(
+          screenSize.width,
+        ),
+      ),
+      child: Row(children: <Widget>[
+        Expanded(
+          child: TextButton(
+            style: TextButton.styleFrom(
+              backgroundColor: SmoothTheme.getColor(
+                Theme.of(context).colorScheme,
+                SmoothTheme.MATERIAL_COLORS[SmoothTheme.COLOR_TAG_BLUE]!,
+                ColorDestination.BUTTON_BACKGROUND,
+              ),
+              shape: RoundedRectangleBorder(
+                borderRadius: BorderRadius.circular(SMALL_SPACE),
+              ),
+              primary: Colors.white,
+            ),
+            onPressed: () {
+              OnboardingFlowNavigator(UserPreferences.getUserPreferences())
```

Comment:
`UserPreferences.getUserPreferences()` is async. It would probably be cleaner to use the provider, like in the rest of the code. Or to pass `userPreferences` as a parameter in the contructor.

Reply:
didn't know `userPreferences` was available on the context. Done